src/lib/homestore_backend/hs_shard_manager.cpp

```diff
@@ -102,8 +102,54 @@ ShardManager::AsyncResult< ShardInfo > HSHomeObject::_create_shard(pg_id_t pg_ow
     });
 }
 
-ShardManager::Result< ShardInfo > HSHomeObject::_seal_shard(shard_id_t id) {
-    return folly::makeUnexpected(ShardError::UNKNOWN_SHARD);
+ShardManager::AsyncResult< ShardInfo > HSHomeObject::_seal_shard(shard_id_t shard_id) {
+    pg_id_t pg_id;
+    ShardInfo shard_info;
+    {
+        std::scoped_lock lock_guard(_shard_lock);
+        auto shard_iter = _shard_map.find(shard_id);
+        RELEASE_ASSERT(shard_iter != _shard_map.end(), "shard id not found");
+        auto hs_shard = d_cast< HS_Shard* >((*shard_iter->second).get());
+        shard_info = hs_shard->info;
+        pg_id = hs_shard->info.placement_group;
+    }
```

Comment:
We already called `get_shard` in `lib/shard_manager.cpp`; we should use its result not take the lock all over again. Probably means taking a `shard_it` rather than a `shard_id_t`.

Reply:
Will change the internal API to take a const ShardInfo&